src/Database/Statement/PDOStatement.php

```diff
@@ -114,8 +115,9 @@ public function fetch($type = 'num')
      * @param string $type num for fetching columns as positional keys or assoc for column names as keys
      * @return array list of all results from database for this statement
      */
-    public function fetchAll($type = 'num')
+    public function fetchAll($type = null)
     {
+        $type = $this->getFetchMode($type);
```

Comment:
Could these be `$type = $type ?: $this->getFetchMode();` This makes the getFetchMode() method a simple getter, and avoids additional function calls for the primary use cases in the ORM.